index.js

```diff
@@ -580,6 +593,150 @@ module.exports = {
 
     self.rewriteUrl = url => {
       return url;
-    }
+    };
+
+    self.getPagesTree = async (req) => {
+      self.maps = {};
+      const excludedTypes = [
+        'workflow-document',
+        ...self.options.excludeTypes || []
+      ]
+
+      const pages = await getPages();
+      const pagesWithPieces = await getPieces(pages);
+      const pagesTree = buildPagesTree(pagesWithPieces);
+
+      return pagesTree;
+
+      async function getPages () {
+        try {
+          const pages = await self.apos.pages
+            .find(req, {}, {
+              _id: 1,
+              title: 1,
+              slug: 1,
+              path: 1,
+              level: 1,
+              rank: 1,
+              _url: 1
+            })
+            .areas(false)
+            .joins(false)
+            .sort({
+              level: -1,
+              rank: -1
+            })
+            .toArray();
+
+          return pages
+            .filter((page) => !excludedTypes.includes(page.type))
+            .map((page) => ({
+              ...page,
+              _children: []
+            }));
+        } catch (err) {
+          self.apos.utils.error(err);
+        }
+      }
+
+      async function getPieces (pages) {
```

Comment:
It is hard to tell if there are two redundant sets of logic now for getting pieces due to the refactoring, but please make sure they are not.

Reply:
Not that much, I already created the `getPiecesModules` to be reused, I also use `self.findPieces` that was already existing. but for the rest the classic getPieces methods write in self.maps and add extra properties and operations that we don't need.